### ISSUE TITLE
[BUG] VS Code selecting wrong Aspire template version by default

### ISSUE THREAD

**Issue by claudiaregio** [microsoft/vscode-dotnettools#1643]
### Describe the Issue

I do not have the .NET 9 SDK, but updating to the latest versions of the Aspire templates leads to VS Code picking the .NET 9 versions by default. Leads me to having "Failed to restore NuGet packages for the solution" error and putting me in a bad state. 

![Image](https://github.com/user-attachments/assets/5bb2f12d-4ae8-431f-816e-922d93c4de7c)
![Image](https://github.com/user-attachments/assets/25580c0c-9b67-4274-bd29-eefc6d631de6)

### Steps To Reproduce

1. Have only .NET 8 installed 
2. Run `dotnet new install Aspire.ProjectTemplates`
3. Create new project in VS Code-> .NET Aspire Starter App -> Click "Show all template options" 
4. Default that is selected is .NET 9 SDK but I don't have the .NET 9 SDK. 

### Expected Behavior

For the default to be the latest version based on the SDKs I actually have installed 

### Environment Information

Apple M2 Pro 
Sonoma 14.2.1
VS Code Version: 1.95.3
C# Dev Kit: v.1.13.9

**Comment by timheuer** [comment]
Thanks...I think we knew this might be a situation with the new template aspect. @damianedwards wasn't this one of the scenarios we knew might happen without the proper TFM check on the template engine?

**Comment by DamianEdwards** [comment]
Yep. The templates default the TFM to `net9.0` so if you're not using the .NET 9 SDK and you don't select the TFM `net8.0` you'll get errors. Unfortunate trade-off we have to make for now.

**Comment by timheuer** [comment]
I'm inclined to close this here so we don't do something only VSCode-specific. For the templates we wanted the template engine to pivot off of TFM and it feels proper that any resolution surface there rather than try any workaround in this extension.  I do think this will be a low hitter for folks -- meaning if you've gone through the effort to get the new templates you likely have the proper matching SDK.

**Comment by claudiaregio** [comment]
@timheuer If you close this one where are you going to reopen the issue? I'm not sure I agree with the low hitter comment since I think anyone who wants to use LTS and VS Code will run into this while following the instructions... 

Is the quickest fix here to help users avoid a bad state to update the docs (https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/aspire-sdk-templates?pivots=dotnet-cli) and have them install targeted versions by default? As opposed to the current instructions which pull both versions by default? 

If they plan to use .NET 8 -> `dotnet new install Aspire.ProjectTemplates::8.0.0`
If they plan to use .NET 9 -> `dotnet new install Aspire.ProjectTemplates::9.0.0`

**Comment by DamianEdwards** [comment]
@claudiaregio the template package version does not map to the version of Aspire the included templates use for lots of reasons. You need to use the latest version of the templates package to get the option to create Aspire 9 projects that target .NET 8.

That said, we can certainly improve documentation to try and make it clearer what's needed to create Aspire 9 projects on .NET 8
